tests/integration/ha/test_storage.py

```diff
@@ -148,7 +209,7 @@ async def test_storage_reuse_in_new_cluster_after_app_removal(
         storage_ids.append(storage_id(ops_test, app, unit_id))
 
     # remove application
-    await ops_test.model.applications[app].destroy()
+    await ops_test.model.applications[app].destroy(force=True, no_wait=True)
```

Comment:
app with `force=True` does not mean "removing machines with force=True". This is important because we have currently #243, which does not allow us to cleanly remove the app with a single `remove-application` command.

I'd recommend you replace this for [this logic](https://github.com/canonical/opensearch-operator/blob/4cc224a57d22f83ce4dee13ef3695319e151e2b8/tests/integration/ha/test_backups.py#L580-L588). Remove the machines one-by-one, then remove the app.

Also, you should wait for it (`no_wait=False`).

Reply:
Thanks for providing that input, I've seen that error too and didn't fully understand yet the reason why removal failed. That's why I've skipped this test.

Reply:
adjusted the logic